build.zig

```diff
@@ -38,7 +38,8 @@ pub fn build(b: *Builder) !void {
     exec.setOutputDir(b.cache_root);
     exec.addBuildOption(bool, "rt_test", rt_test);
     exec.setBuildMode(build_mode);
-    exec.setLinkerScriptPath("link.ld");
+    const linker_script_path = try fs.path.join(b.allocator, &[_][]const u8{ "src/kernel/arch", target_str, "link.ld" });
```

Comment:
The string `"src/kernel/arch"` is used for the constants, so maybe can refactor this out

Reply:
Nice idea